### PR TITLE
update minimal checkpointTS as gc safepoint to PD

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve? 

close https://github.com/pingcap/ticdc/issues/482.

### What is changed and how it works?

update minimal checkpointTS as gc safepoint to PD

Tests 

 - No...

Code changes

 - Has interface methods change

Side effects

 - 

Related changes

 -

### CODE REVIEW

File: cdc/owner.go [hunk 1]
```diff
@@ -295,10 +296,19 @@ func (o *Owner) loadChangeFeeds(ctx context.Context) error {
...
+		if changefeed.status.CheckpointTs < minCheckpointTs {
+			minCheckpointTs = changefeed.status.CheckpointTs
+		}
+	}
+	err := o.cfRWriter.PutAllChangeFeedStatus(ctx, snapshot)
+	if err != nil {
+		return errors.Trace(err)
 	}
-	return errors.Trace(o.cfRWriter.PutAllChangeFeedStatus(ctx, snapshot))
+	_,err = o.pdClient.UpdateGCSafePoint(ctx, minCheckpointTs)
```
Comment (overvenus) [inline cdc/owner.go:310]: I think we should call [`UpdateServiceGCSafePoint`](https://github.com/pingcap/pd/blob/7bef70bee8a9fc59af741576328de0a4ecc15302/client/client.go#L74) ? @MyonKeminta

File: cdc/owner.go [hunk 2]
```diff
@@ -295,10 +298,23 @@ func (o *Owner) loadChangeFeeds(ctx context.Context) error {
...
+		if changefeed.status.CheckpointTs < minCheckpointTs {
+			minCheckpointTs = changefeed.status.CheckpointTs
+		}
+	}
+	err := o.cfRWriter.PutAllChangeFeedStatus(ctx, snapshot)
+	if err != nil {
+		return errors.Trace(err)
 	}
-	return errors.Trace(o.cfRWriter.PutAllChangeFeedStatus(ctx, snapshot))
+	_, err = o.pdClient.UpdateServiceGCSafePoint(ctx, cdcServiceSafePointID, math.MaxInt64, minCheckpointTs)
```
Comment (overvenus) [inline cdc/owner.go:312]: Could you add a TODO about TTL needs to be configurable? As setting TTL to MaxInt64 may not be acceptable in some scenarios.

**Comment by leoppro** [comment]
please resolve the conflicts

File: cdc/owner.go [hunk 3]
```diff
@@ -299,10 +302,23 @@ func (o *Owner) balanceTables(ctx context.Context) error {
 
 func (o *Owner) flushChangeFeedInfos(ctx context.Context) error {
 	snapshot := make(map[model.ChangeFeedID]*model.ChangeFeedStatus, len(o.changeFeeds))
+	minCheckpointTs := uint64(math.MaxUint64)
 	for id, changefeed := range o.changeFeeds {
```
Comment (leoppro) [inline cdc/owner.go:306]: what happened if it doesn't exist changefeed

Comment (crazycs520) [reply cdc/owner.go:313]: Great catch.

**Pull request merged**